src/audio/dai-legacy.c

```diff
@@ -317,12 +317,15 @@ static int dai_comp_hw_params(struct comp_dev *dev,
 	return 0;
 }
 
-static int dai_verify_params(struct comp_dev *dev,
+static int dai_verify_params(struct dai_data *dd, struct comp_dev *dev,
 			     struct sof_ipc_stream_params *params)
 {
 	struct sof_ipc_stream_params hw_params;
+	int ret;
 
-	comp_dai_get_hw_params(dev, &hw_params, params->direction);
+	ret = dai_zephyr_get_hw_params(dd, dev, &hw_params, params->direction);
```

Comment:
@btian1 @ranj063 Hmm. This is getting very confusing to read. Why do we use "dai_zephyr_" name for both legacy and Zephyr? This seems to have started in 79c43394892e05bb4778c7b30c87a007dd41d3be . I mean, dai-legacy.c is specifically NOT using Zephyr DAI interface, so it makes the code very hard to read. FYI @juimonen

Reply:
@kv2019i yeah, it is confusing, but it's what we have both in host-{zephyr,legacy}.c and in dai-{zephyr,legacy}.c.

Reply:
@kv2019i it is confusing but the copier could in theory use either dai-legacy or dai-zephyr and this common function name allows us to unify the copier for both cases. Maybe dai_zephyr_ isnt the best prefix. Should we just remove the zephyr from the name and replace it with dai_common instad of dai_zephyr?

Reply:
let me remove zephyr prefix and submit with a new patchset

Reply:
Let's handle in a follow-up.